modules/gitops-argo-rollouts-architecture-overview.adoc

```diff
@@ -0,0 +1,70 @@
+|===
+
+Argo Rollouts also supports percentage-based traffic routing for canary-based deployments. Rollouts integrate with the Kubernetes cluster networking and routing resources, which are referred to as traffic management.
+
+[options="header"]
+|===
+|Name | Description
+|`Services / Ingresses` | Rollouts natively support routing traffic by Services and Ingresses, using the Service and Ingress controllers.
+|`OpenShift Route resources` | Rollouts perform traffic splitting across different application versions using OpenShift Route resources.
+|`OpenShift Service Mesh's VirtualService resources` | Rollouts also perform traffic splitting across different versions of an Application, using OpenShift Service Mesh with Gateways VirtualServices resources.
+|===
+
+Argo Rollouts include several mechanisms for gathering analysis metrics for verifying that a new version of an application is deployed.
+
+[options="header"]
+|===
+|Name | Description
+|`Prometheus metrics` | `AnalysisTemplate` is configured to connect to Prometheus instances, to evaluate the success or failure of one or more metrics.
+|`Kubernetes Job resource metrics` | Rollouts support the use of simple Kubernetes jobs to run analysis on resource metrics. You can verify a successful deployment of an application based on the successful execution of Kubernetes jobs.
+|===
+
+[id="gitops-components_{context}"]
+== Components
+
+Argo Rollouts consists of several components that enable users to practice Progressive Delivery on OpenShift.
+
+[options="header"]
+|===
+|Name | Description
+|`Rollouts Controller` | The Rollouts Controller component of the Argo Rollouts manages the Rollouts custom resource which resembles a native Kubernetes Deployment resource. The Rollouts Controller does not modify the standard deployment resources and only responds to changes in the Rollouts resources.
+|`AnalysisRun Controller` | The `AnalysisRun` controller performs analysis for an `AnalysisRun` resource and manages the `AnalysisRun` custom resource. It provides the capability to connect a Rollout to your metrics provider and define specific thresholds for certain metrics that decide if a deployment update is successful for your application.
+
+Argo Rollouts supports two custom Kubernetes resources to perform Analysis - `AnalysisTemplate` and `AnalysisRun`. You can define an `AnalysisTemplate` on a specific Rollout or globally on a cluster to be shared by multiple rollouts as a Cluster`AnalysisTemplate`. The `AnalysisRun` resource is scoped on a specific rollout.
+|`Experiment Controller` | The `Experiment` Controller creates analysis for limited-duration, short-lived replica sets, and manages the `Experiment` custom resource. The `Experiment` custom resource can be used independently or executed as part of Rollouts. You can link a list of `AnalysisTemplate` to be used on replica sets by creating an `Experiment` custom resource.
```

Comment:
```suggestion
|Experiment controller |The Experiment controller creates analysis for short-lived replica sets, and manages the `Experiment` CR. 
```

I have some questions/suggestions regarding this one:

1) " limited-duration, short-lived" : Do  these mean the same thing? Or is there some significant difference?

2) The sentence: " The `Experiment` CR can be used independently or executed as part of Rollouts."

     This would make more sense to include in the  `Experiment` CR entry in the table resource.
Just add it with this small change: "The `Experiment` CR can be used independently or run as part of rollouts."

3)  For the `Experiment` CR you mentioned this: 
    * "The Experiments resource is used to run short, limited-duration analysis on an application" 

    And in this section it is mentioned this: 
    * "The `Experiment` Controller creates analysis for limited-duration, short-lived replica sets"

    So now I am confused, are the analyses short and time-limited, or are the replica sets, for which the analyses are made, short and time-limited? Or is both true? 

    Also, based on this, does this mean that the controller creates analysis for replica sets, while the CR creates it for applications (not sure what the difference here might be as I suppose the replica set here means a version of the application)? (Or is it possibly that the controller creates analyses while the CR is used to actually run them?) This is not clear to me based on these descriptions.

4) The sentence: "You can link `AnalysisTemplate` CRs to be used on replica sets by creating an `Experiment` custom resource."
This would make more sence to include in the description of the suggested `AnalysisTemplate` row.

Reply:
@jgwest - FYI.

Reply:
>  " limited-duration, short-lived" : Do these mean the same thing? Or is there some significant difference?

They mean the same thing , not sure why I included both :grin:

> So now I am confused, are the analyses short and time-limited, or are the replica sets, for which the analyses are made, short and time-limited? Or is both true?

Both are true:
- The analysis is run for a short period of time
- The analysis is run on the replicas sets, which likewise exist for a short period of time

The lifecycle of the experiment, the analysis, and the replicasets, are equal:
- when the experiment starts, the replicasets are created.
- once the experiment ends, the analysis is complete, and the replicaset is deleted.

> Also, based on this, does this mean that the controller creates analysis for replica sets, while the CR creates it for applications (not sure what the difference here might be as I suppose the replica set here means a version of the application)? (Or is it possibly that the controller creates analyses while the CR is used to actually run them?) This is not clear to me based on these descriptions.

In this case, the Rollouts controller only creates ReplicaSets (there is no concept of application is Rollouts, except as a general concept to refer to whatever the user is deploying). 

When an Experiment CR is created, the Experiment controller:
1) Creates 1 or more replicasets based on the contents of the CR.
2) Runs an analysis on those replicasets
3) Once the analysis completes:
    -  the .status of the Experiment is updated with the result
    - the replicasetes are deleted.

Reply:
Amazing, thank you @jgwest for the explanation! 

@Dhruv-Soni11 feel free to add the last change recommeded by SME here :+1:  
I will have some additonal sugestions regarding this later. (Depending on the time constraint here, we can spply the additional suggestions, or add them together with the enhancement Jira you mentioned that would come after the release.)

Reply:
Fixed.